code/modules/mob/living/status_procs.dm

```diff
@@ -168,6 +185,8 @@
 		status_traits -= trait
 
 /mob/living/proc/has_trait(trait, list/sources)
+	if(roundstart_traits[trait])
+		return roundstart_traits[trait] //fetch the trait object directly in case we need to modify it
```

Comment:
This really shouldn't have different return types depending if you check one define or another